SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java

```diff
@@ -190,4 +206,88 @@ private void updateLongRunningTasksUsage(Map<ResourceUsageType, Number> longRunn
     longRunningTasksUsage.compute(ResourceUsageType.MEMORY_BYTES_USED, (k, v) -> (v == null) ? memBytesUsed : v.longValue() + memBytesUsed);
     longRunningTasksUsage.compute(ResourceUsageType.CPU_USED, (k, v) -> (v == null) ? cpuUsed : v.doubleValue() + cpuUsed);
   }
+
+  private SingularityClusterUtilization getClusterUtilization(Map<String, RequestUtilization> utilizationPerRequestId, double minUnderUtilizedPct, long totalMemBytesUsed, long totalMemBytesAvailable, double totalCpuUsed, double totalCpuAvailable, long now) {
+    int numRequestsWithUnderUtilizedCpu = 0;
+    int numRequestsWithOverUtilizedCpu = 0;
+    int numRequestsWithUnderUtilizedMemBytes = 0;
+
+    double totalUnderUtilizedCpu = 0;
+    double totalOverUtilizedCpu = 0;
+    long totalUnderUtilizedMemBytes = 0;
+
+    double maxUnderUtilizedCpu = Double.MIN_VALUE;
+    double maxOverUtilizedCpu = Double.MIN_VALUE;
+    long maxUnderUtilizedMemBytes = Long.MIN_VALUE;
+
+    double minUnderUtilizedCpu = Double.MAX_VALUE;
+    double minOverUtilizedCpu = Double.MAX_VALUE;
+    long minUnderUtilizedMemBytes = Long.MAX_VALUE;
+
+
+    for (Iterator<Entry<String, RequestUtilization>> it = utilizationPerRequestId.entrySet().iterator(); it.hasNext(); ) {
+      RequestUtilization utilization = it.next().getValue();
+      Optional<SingularityDeploy> maybeDeploy = deployManager.getDeploy(utilization.getRequestId(), utilization.getDeployId());
+
+      if (maybeDeploy.isPresent() && maybeDeploy.get().getResources().isPresent()) {
+        boolean includeUtilization = true;
+        long memoryBytesReserved = (long) (maybeDeploy.get().getResources().get().getMemoryMb() * SingularitySlaveUsage.BYTES_PER_MEGABYTE);
+        double cpuReserved = maybeDeploy.get().getResources().get().getCpus();
+
+        double unusedCpu = cpuReserved - utilization.getAvgCpuUsed();
+        long unusedMemBytes = memoryBytesReserved - utilization.getMemBytesTotal();
+
+        if (unusedCpu / cpuReserved >= minUnderUtilizedPct) {
+          numRequestsWithUnderUtilizedCpu++;
+          totalUnderUtilizedCpu += unusedCpu;
+          maxUnderUtilizedCpu = Math.max(unusedCpu, maxUnderUtilizedCpu);
+          minUnderUtilizedCpu = Math.min(unusedCpu, minUnderUtilizedCpu);
+        } else if (unusedCpu < 0) {
+          double overusedCpu = Math.abs(unusedCpu);
+
+          numRequestsWithOverUtilizedCpu++;
+          totalOverUtilizedCpu += overusedCpu;
+          maxOverUtilizedCpu = Math.max(overusedCpu, maxOverUtilizedCpu);
+          minOverUtilizedCpu = Math.min(overusedCpu, minOverUtilizedCpu);
+        } else {
+          includeUtilization = false;
+        }
+
+        if (unusedMemBytes / memoryBytesReserved >= minUnderUtilizedPct) {
```

Comment:
I added this min percentage to give us a goal of where we want our utilization to be. It's defaulted to 5% so that means any task using less than 95% of resources allocated would be considered in our report. 

Without a min percent, it'd skew the counts and averages. Every task not using all the requested resources, no matter how small, will be counted as under-utilized. If some task is using 98 mb out of the 100 mb they requested, that seems very reasonable so it didn't make sense to count it as an under-utilized task.

With that said, I'm okay with changing it to consider all tasks if you think that would be more useful